### PR TITLE
Rework topology-view to use shared data

### PR DESCRIPTION
**Pull request by rvsia**
part of https://issues.redhat.com/browse/RHCLOUD-9341

- data is shared between topology/treeview
- simpler code
- allows to add new structures to topology without editing the code generating it

image

demo

### CODE REVIEW

**Comment by Hyperkid123** [comment]
@rvsia I expect that we will need to do some modification to the topology component itself. Do you have any idea what will be required in a near future?

**Review by Hyperkid123 — APPROVED** [review]

File: src/api/topology-viewer-api.js [hunk 1]
```diff
@@ -5,52 +5,83 @@ const API = getAxtionsInstace();
...
+        nodeType: 'serviceInstanceNodes',
+      },
     },
     {
       name: 'service_offering_nodes',
       label: 'Service offering nodes',
       attributes: ['name'],
+      topologyViewNode: {
+        nodeShape: 'square',
+        nodeType: 'serviceOfferingsNode',
```
Comment (slemrmartin) [inline src/api/topology-viewer-api.js:66]: @rvsia isn't there missing "s"? :)

Comment (slemrmartin) [reply src/api/topology-viewer-api.js:66]: or one extra "s" and one missing "s"

Comment (rvsia) [reply src/api/topology-viewer-api.js:66]: Yeah, but it's also missing "s" in the iconMapper so it actually works as it should. :smile:

Comment (Hyperkid123) [reply src/api/topology-viewer-api.js:66]: :mage_man:

**Pull request merged**